File: estDados_python/aula_13_04/dado.py
class Dado:
    def __init__(self, valor, linha, coluna):
        self.valor = valor
        self.linha = linha
        self.coluna = coluna

    def __eq__(self, other):
        if not isinstance(other, Dado):
            return False
        return self.linha == other.linha and self.coluna == other.coluna
    
    def __str__(self):
        return f"Dado [Valor={self.valor}, Linha={self.linha}, Coluna={self.coluna}]"
    

File: estDados_Java/aula_13_04/Dado.java
package aula_13_04;

public class Dado {
    public int valor;
    public int linha;
    public int coluna;

    public Dado(int valor, int linha, int coluna) {
        this.valor = valor;
        this.linha = linha;
        this.coluna = coluna;
    }


    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Dado other = (Dado) obj;
        if (linha != other.linha)
            return false;
        if (coluna != other.coluna)
            return false;
        return true;
    }


    @Override
    public String toString() {
        return "Dado [valor=" + valor + ", linha=" + linha + ", coluna=" + coluna + "]";
    }

    
}
